pd-client/client.go

```diff
@@ -43,77 +50,271 @@ type Client interface {
 	Close()
 }
 
+type tsoRequest struct {
+	done     chan error
+	physical int64
+	logical  int64
+}
+
+const (
+	pdTimeout             = 3 * time.Second
+	maxMergeTSORequests   = 10000
+	maxInitClusterRetries = 100
+)
+
 type client struct {
-	worker *rpcWorker
+	urls        []string
+	clusterID   uint64
+	tsoRequests chan *tsoRequest
+
+	mu            sync.RWMutex
```

Comment:
I see no other files in this repo is using this style, maybe change them all in another PR later.